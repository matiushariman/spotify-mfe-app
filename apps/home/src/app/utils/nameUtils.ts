export const displayTrackName = (trackName: string) => {
  const MAX_TRACK_NAME_LENGTH = 15;

  if (trackName.length <= MAX_TRACK_NAME_LENGTH) {
    return trackName;
  }

  return `${trackName.slice(0, MAX_TRACK_NAME_LENGTH)}...`;
};

export const displayArtistName = (artistName: string) => {
  const MAX_ARTIST_NAME_LENGTH = 23;

  if (artistName.length <= MAX_ARTIST_NAME_LENGTH) {
    return artistName;
  }

  return `${artistName.slice(0, MAX_ARTIST_NAME_LENGTH)}...`;
};
