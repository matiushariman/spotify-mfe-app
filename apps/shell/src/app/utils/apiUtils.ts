import type { ApiErrorResponse } from '@spotify-mfe-app/api-client';
import { removeAuthToken } from '@spotify-mfe-app/auth-utils';

export const handleApiError = (err: ApiErrorResponse) => {
  const {
    error: { status },
  } = err;

  if (status === 401) {
    window.alert(
      'Your session has expired! Click close to reauthenticate your session.'
    );

    removeAuthToken();
    window.location.replace('/');
  }
};
