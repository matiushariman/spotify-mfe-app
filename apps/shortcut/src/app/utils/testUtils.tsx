import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClientProvider, QueryClient } from 'react-query';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={new QueryClient()}>
    {children}
  </QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render, AllTheProviders as wrapper };
