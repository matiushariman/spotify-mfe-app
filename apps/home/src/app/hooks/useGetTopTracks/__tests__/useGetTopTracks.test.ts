import { useGetTopTracks } from '..';

import { renderHook, wrapper, waitFor } from '../../../utils/testUtils';

describe('components/hooks/useGetTopTracks', () => {
  it('should return top tracks response', async () => {
    const { result } = renderHook(() => useGetTopTracks(), {
      wrapper,
    });

    await waitFor(() => expect(result.current.isSuccess).toBe(true));

    expect(result.current.data).toMatchSnapshot();
  });
});
