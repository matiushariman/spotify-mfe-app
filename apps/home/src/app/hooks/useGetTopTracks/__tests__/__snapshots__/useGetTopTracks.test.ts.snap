// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/hooks/useGetTopTracks should return top tracks response 1`] = `
Object {
  "href": "https://api.spotify.com/v1/me/shows?offset=0&limit=20
",
  "items": Array [
    Object {
      "album": Object {
        "album_type": "ALBUM",
        "artists": Array [
          Object {
            "external_urls": Object {
              "spotify": "https://open.spotify.com/artist/2iDVt6mFbtbDEZG5ax0dTi",
            },
            "href": "https://api.spotify.com/v1/artists/2iDVt6mFbtbDEZG5ax0dTi",
            "id": "2iDVt6mFbtbDEZG5ax0dTi",
            "name": "Tulus",
            "type": "artist",
            "uri": "spotify:artist:2iDVt6mFbtbDEZG5ax0dTi",
          },
        ],
        "available_markets": Array [
          "AD",
          "AE",
          "AR",
          "AT",
          "AU",
          "BE",
          "BG",
          "BH",
          "BO",
          "BR",
          "CA",
          "CH",
          "CL",
          "CO",
          "CR",
          "CY",
          "CZ",
          "DE",
          "DK",
          "DO",
          "DZ",
          "EC",
          "EE",
          "EG",
          "ES",
          "FI",
          "FR",
          "GB",
          "GR",
          "GT",
          "HK",
          "HN",
          "HU",
          "ID",
          "IE",
          "IL",
          "IN",
          "IS",
          "IT",
          "JO",
          "JP",
          "KW",
          "LB",
          "LI",
          "LT",
          "LU",
          "LV",
          "MA",
          "MC",
          "MT",
          "MX",
          "MY",
          "NI",
          "NL",
          "NO",
          "NZ",
          "OM",
          "PA",
          "PE",
          "PH",
          "PL",
          "PS",
          "PT",
          "PY",
          "QA",
          "RO",
          "SA",
          "SE",
          "SG",
          "SK",
          "SV",
          "TH",
          "TN",
          "TR",
          "TW",
          "US",
          "UY",
          "VN",
          "ZA",
        ],
        "external_urls": Object {
          "spotify": "https://open.spotify.com/album/3R4IAF9ApqYeUQrv1ddyoR",
        },
        "href": "https://api.spotify.com/v1/albums/3R4IAF9ApqYeUQrv1ddyoR",
        "id": "3R4IAF9ApqYeUQrv1ddyoR",
        "images": Array [
          Object {
            "height": 640,
            "url": "https://i.scdn.co/image/ab67616d0000b273b55d26c578e30129b0a7e86e",
            "width": 640,
          },
          Object {
            "height": 300,
            "url": "https://i.scdn.co/image/ab67616d00001e02b55d26c578e30129b0a7e86e",
            "width": 300,
          },
          Object {
            "height": 64,
            "url": "https://i.scdn.co/image/ab67616d00004851b55d26c578e30129b0a7e86e",
            "width": 64,
          },
        ],
        "name": "Manusia",
        "release_date": "2022-03-03",
        "release_date_precision": "day",
        "total_tracks": 10,
        "type": "album",
        "uri": "spotify:album:3R4IAF9ApqYeUQrv1ddyoR",
      },
      "artists": Array [
        Object {
          "external_urls": Object {
            "spotify": "https://open.spotify.com/artist/2iDVt6mFbtbDEZG5ax0dTi",
          },
          "href": "https://api.spotify.com/v1/artists/2iDVt6mFbtbDEZG5ax0dTi",
          "id": "2iDVt6mFbtbDEZG5ax0dTi",
          "name": "Tulus",
          "type": "artist",
          "uri": "spotify:artist:2iDVt6mFbtbDEZG5ax0dTi",
        },
      ],
      "available_markets": Array [
        "AD",
        "AE",
        "AR",
        "AT",
        "AU",
        "BE",
        "BG",
        "BH",
        "BO",
        "BR",
        "CA",
        "CH",
        "CL",
        "CO",
        "CR",
        "CY",
        "CZ",
        "DE",
        "DK",
        "DO",
        "DZ",
        "EC",
        "EE",
        "EG",
        "ES",
        "FI",
        "FR",
        "GB",
        "GR",
        "GT",
        "HK",
        "HN",
        "HU",
        "ID",
        "IE",
        "IL",
        "IN",
        "IS",
        "IT",
        "JO",
        "JP",
        "KW",
        "LB",
        "LI",
        "LT",
        "LU",
        "LV",
        "MA",
        "MC",
        "MT",
        "MX",
        "MY",
        "NI",
        "NL",
        "NO",
        "NZ",
        "OM",
        "PA",
        "PE",
        "PH",
        "PL",
        "PS",
        "PT",
        "PY",
        "QA",
        "RO",
        "SA",
        "SE",
        "SG",
        "SK",
        "SV",
        "TH",
        "TN",
        "TR",
        "TW",
        "US",
        "UY",
        "VN",
        "ZA",
      ],
      "disc_number": 1,
      "duration_ms": 242000,
      "explicit": false,
      "external_ids": Object {
        "isrc": "FR2X42253130",
      },
      "external_urls": Object {
        "spotify": "https://open.spotify.com/track/2hHeGD57S0BcopfVcmehdl",
      },
      "href": "https://api.spotify.com/v1/tracks/2hHeGD57S0BcopfVcmehdl",
      "id": "2hHeGD57S0BcopfVcmehdl",
      "is_local": false,
      "name": "Hati-Hati di Jalan",
      "popularity": 80,
      "preview_url": "https://p.scdn.co/mp3-preview/c5e70e742b9f7c131676f6cfb9c69e5fc4a4f85b?cid=06a2a95448cb464d870d7d98373601d9",
      "track_number": 8,
      "type": "track",
      "uri": "spotify:track:2hHeGD57S0BcopfVcmehdl",
    },
  ],
  "limit": 20,
  "next": "https://api.spotify.com/v1/me/shows?offset=1&limit=1",
  "offset": 0,
  "previous": "https://api.spotify.com/v1/me/shows?offset=1&limit=1",
  "total": 4,
}
`;
